### PR TITLE
[READY]Da food update. Donuts! New foods! Nuggbox! Bowl resprite.

### PR DESCRIPTION
**Pull request by Krysonism**
## About The Pull Request

This PR adds a variety of frosted donuts, a new bowl sprite and a chicken nugget box. This PR also adds 3 new dishes using my new plants.
 
Bungo curry - a vegetable curry, sadly we don't have coconut milk yet, so it is not appropriate for vegans.
Spaceman's cake - a purple cake made using spaceman's trumpet flowers.
Electron soup - An ethereal food with an animated sprite.

I sadly had to delete the cherry jelly donuts from the game, the reason I did this was to reduce crafting menu bloat and lag.

I could let people make, but not frost them, but this would be confusing.

I have also renamed the frost proc to expand its use to other types of donut decoration.

I  also adjusted tea powder to be less toxic and better tasting.

## Why It's Good For The Game

More foods are always fun.

I have attempted to add some foods cooked using the new plants I've added a while back, plants should have some applications beyond powergaming reagents.

The new bowl sprite is in my opinion superior, but I understand if you want to keep old sprite, let me know in the comments if you want to ditch the old sprite.

The frosted donuts are a good addition because they are desirable, colorful and the concept can be extended with many other variations if desired. 

All the frosted donuts also have more sprinkles than the plain donuts, and some donuts can even have 3 sprinkles. Security buff!

Nuggets can take up a lot of counter space
And this is why I've added a box that lets you present them to the diners in an attractive manner. It is also good if you want to do a fast food gimmick.
  

donut_comp
bungo_curry_comparasion
cake_slice_trumpet_comparasion
trumpet_cake_comparasion
bowlcomparasion
nuggbox
electronsoup64

## Changelog
:cl:
add: Added lots of new donuts  
add: 3 new dishes cooked with the new plants.
add: Added a nugget box. I'm lovin' it
del: You can no longer make cherry donuts.
tweak: Tea powder is less toxic and tastes better.
/:cl:

### CODE REVIEW

**Comment by carshalash** [comment]
Nice sprites!

**Comment by Krysonism** [comment]
I will update the PR with more information once i get to work.

**Comment by Potato-Masher** [comment]
Blessed bowl sprite, actually looks like a bowl.

**Comment by GuyonBroadway** [comment]
Rename the nugbox to be a box of nugs so its a play on the box of hugs the clown spawns with.

**Comment by ArcaneMusic** [comment]
Not only does the food look better, but the bowl looks like it was WASHED

**Comment by Farquaar** [comment]
>He removed cherry jelly donuts
Why even live anymore?

**Comment by PKPenguin321** [comment]
>Rename the nugbox to be a box of nugs so its a play on the box of hugs the clown spawns with.

Do you realize the box of hugs is a reference to the phrase "hugbox," which "nugbox" is already seemingly a parody of

**Comment by GuyonBroadway** [comment]
>Do you realize the box of hugs is a reference to the phrase "hugbox," which "nugbox" is already seemingly a parody of


Very much so but this game of chinese whispers could end up somewhere funny.

Box of bugs up next.

**Comment by Krysonism** [comment]
I have done some more testing and have noticed two bugs. 

Electron soup does not delete it's ingredients upon crafting

The plain donut recipe shows up twice in the crafting menu

**Comment by 81Denton** [comment]
Merge conflicts ahead

**Comment by Krysonism** [comment]
I will resolve conflict and take another look at this tomorrow

**Comment by Krysonism** [comment]
This compiles and crafting the electron soup seems to work. It should be ready for merge.

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 1]
```diff
@@ -384,7 +384,8 @@ GLOBAL_LIST_INIT(cardboard_recipes, list (														\
 		new /datum/stack_recipe("donut box", /obj/item/storage/fancy/donut_box),				\
 		new /datum/stack_recipe("egg box", /obj/item/storage/fancy/egg_box),					\
 		new /datum/stack_recipe("donk-pockets box", /obj/item/storage/box/donkpockets),			\
-		new /datum/stack_recipe("monkey cube box", /obj/item/storage/box/monkeycubes),			\
+		new /datum/stack_recipe("monkey cube box", /obj/item/storage/box/monkeycubes),
+		new /datum/stack_recipe("nuggetbox box", /obj/item/storage/fancy/nugget_box),			\
```
Comment (81Denton) [inline code/game/objects/items/stacks/sheets/sheet_types.dm]:
```suggestion
		new /datum/stack_recipe("nugget box", /obj/item/storage/fancy/nugget_box),			\
```

File: code/game/objects/items/storage/fancy.dm [hunk 2]
```diff
@@ -350,3 +350,18 @@
...
+
+/obj/item/storage/fancy/nugget_box
+	name = "nugget box"
+	desc = "A cardboard box used for holding chicken nuggies."
+	icon = 'icons/obj/food/containers.dmi'
+	icon_state = "nuggetbox"
+	icon_type = "nugget"
+	spawn_type = /obj/item/reagent_containers/food/snacks/nugget
+
+/obj/item/storage/fancy/heart_box/ComponentInitialize()
```
Comment (81Denton) [inline code/game/objects/items/storage/fancy.dm]:
```suggestion
/obj/item/storage/fancy/nugget_box/ComponentInitialize()
```

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 3]
```diff
@@ -56,11 +61,95 @@
...
+	list_reagents = list(/datum/reagent/consumable/nutriment = 3, /datum/reagent/consumable/ketchup = 2)
+	tastes = list("meat" = 1)
+	foodtype = JUNKFOOD | MEAT | GROSS | FRIED | BREAKFAST
+
+/obj/item/reagent_containers/food/snacks/donut/berry
+	name = "pink donut"
+	desc = "Goes great with a soy latte."
+	icon_state = "donut_pink"
+	bonus_reagents = list(/datum/reagent/consumable/berryjuice = 3, /datum/reagent/consumable/sprinkles = 1) //Extra sprinkles to reward frosting
+	is_decorated = FALSE
```
Comment (81Denton) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]: /obj/item/reagent_containers/food/snacks/donut already has `is_decorated` set to FALSE, no need to define it twice.

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 4]
```diff
@@ -56,11 +61,95 @@
...
+	is_decorated = TRUE
+	filling_color = "#D4AD5B"
+
+/obj/item/reagent_containers/food/snacks/donut/choco
+	name = "chocolate donut"
+	desc = "Goes great with a glass of warm milk."
+	icon_state = "donut_choc"
+	bonus_reagents = list(/datum/reagent/consumable/hot_coco = 3, /datum/reagent/consumable/sprinkles = 1) //the coco reagent is just bitter.
+	tastes = list("donut" = 4, "bitterness" = 1)
+	is_decorated = FALSE
```
Comment (81Denton) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]: same here

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 5]
```diff
@@ -71,29 +160,163 @@
...
 		reagents.add_reagent(extra_reagent, 3)
 
+/obj/item/reagent_containers/food/snacks/donut/jelly/plain //use this ingame to avoid inheritance related crafting issues.
+
+/obj/item/reagent_containers/food/snacks/donut/jelly/berry
+	name = "pink jelly donut"
+	desc = "Goes great with a soy latte."
+	icon_state = "jelly_pink"
+	bonus_reagents = list(/datum/reagent/consumable/berryjuice = 3, /datum/reagent/consumable/sprinkles = 1, /datum/reagent/consumable/nutriment/vitamin = 1) //Extra sprinkles to reward frosting
+	is_decorated = FALSE
```
Comment (81Denton) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]: same here

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 6]
```diff
@@ -71,29 +160,163 @@
...
+	is_decorated = TRUE
+	filling_color = "#D4AD5B"
+
+/obj/item/reagent_containers/food/snacks/donut/jelly/choco
+	name = "chocolate jelly donut"
+	desc = "Goes great with a glass of warm milk."
+	icon_state = "jelly_choc"
+	bonus_reagents = list(/datum/reagent/consumable/hot_coco = 3, /datum/reagent/consumable/sprinkles = 1, /datum/reagent/consumable/nutriment/vitamin = 1) //the coco reagent is just bitter.
+	tastes = list("jelly" = 1, "donut" = 4, "bitterness" = 1)
+	is_decorated = FALSE
```
Comment (81Denton) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]: same here

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 7]
```diff
@@ -71,29 +160,163 @@
...
-	foodtype = JUNKFOOD | MEAT | GROSS | FRIED | BREAKFAST
-
+/obj/item/reagent_containers/food/snacks/donut/jelly/slimejelly/plain
+
+/obj/item/reagent_containers/food/snacks/donut/jelly/slimejelly/berry
+	name = "pink jelly donut"
+	desc = "Goes great with a soy latte."
+	icon_state = "jelly_pink"
+	bonus_reagents = list(/datum/reagent/consumable/berryjuice = 3, /datum/reagent/consumable/sprinkles = 1, /datum/reagent/consumable/nutriment/vitamin = 1) //Extra sprinkles to reward frosting
+	is_decorated = FALSE
```
Comment (81Denton) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]: same here

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 8]
```diff
@@ -71,29 +160,163 @@
...
+	is_decorated = TRUE
+	filling_color = "#D4AD5B"
+
+/obj/item/reagent_containers/food/snacks/donut/jelly/slimejelly/choco
+	name = "chocolate jelly donut"
+	desc = "Goes great with a glass of warm milk."
+	icon_state = "jelly_choc"
+	bonus_reagents = list(/datum/reagent/consumable/hot_coco = 3, /datum/reagent/consumable/sprinkles = 1, /datum/reagent/consumable/nutriment/vitamin = 1) //the coco reagent is just bitter.
+	tastes = list("jelly" = 1, "donut" = 4, "bitterness" = 1)
+	is_decorated = FALSE
```
Comment (81Denton) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]: same here

File: code/modules/food_and_drinks/recipes/tablecraft/recipes_pastry.dm [hunk 9]
```diff
@@ -3,24 +3,25 @@
...
+	name = "Chaos donut"
 	reqs = list(
-		/datum/reagent/consumable/sugar = 1,
+		/datum/reagent/consumable/frostoil = 5,
+		/datum/reagent/consumable/capsaicin = 5,
 		/obj/item/reagent_containers/food/snacks/pastrybase = 1
 	)
-	result = /obj/item/reagent_containers/food/snacks/donut
+	result = /obj/item/reagent_containers/food/snacks/donut/chaos
 	subcategory = CAT_PASTRY
```
Comment (81Denton) [inline code/modules/food_and_drinks/recipes/tablecraft/recipes_pastry.dm]: /datum/crafting_recipe/food/donut already has `subcategory` set to CAT_PASTRY, no need to define it twice

Same goes for all the other child datums like /datum/crafting_recipe/food/donut/slimejelly

**Review by 81Denton — CHANGES_REQUESTED** [review]
You just need to remove a few duplicate definitions/fix typos and this is good to go.

File: code/modules/food_and_drinks/food/snacks_pastry.dm (hunk 8 above)
Comment (Krysonism) [reply code/modules/food_and_drinks/food/snacks_pastry.dm]: fixed

**Comment by wesoda25** [comment]
@Krysonism this PR was fucking excellent and by far is my favorite feature ive seen in game since i took a break. bravo keep up the good work.

**Pull request merged**